server/game/cards/events/05/daringrescue.js

```diff
@@ -0,0 +1,44 @@
+const DrawCard = require('../../../drawcard.js');
+
+class DaringRescue extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Return character then have knight gain power',
+            condition: () => this.hasKnightCharacter(),
+            target: {
+                activePromptTitle: 'Select a character to return',
+                cardCondition: card => (
+                    card.location === 'play area' && 
+                    card.controller === this.controller && 
+                    card.getType() === 'character')
+            },
+            handler: context => {
+                this.controller.moveCard(context.target, 'hand');
+                this.game.promptForSelect(this.controller, {
+                    activePromptTitle: 'Select a Knight character',
+                    source: this,
+                    cardCondition: card => (
+                        card.location === 'play area' && 
+                        card.hasTrait('Knight') &&
+                        card.getType() === 'character' && 
+                        card.controller === this.controller),
+                    onSelect: (p, card) => {
+                        card.modifyPower(1);
+                        this.game.addMessage('{0} plays {1} to return {2} to its owner\'s hand then have {3} gain 1 power', 
+                                              this.controller, this, context.target, card);
```

Comment:
As i understand the card, you might use it to return the sole Knight character you've in play. In that case, the event is playable, but you'll have no Knight to gain power in the second step. I think this card implementation would hang at that point, forcing the user to /cancel-prompt, isn't it?

Aside from that, the message should probably be made smarter to 1) differentiate the two cases, and 2) make sure it get posted when you return the last remaining Knight to hand.